src/modules/cpu/kernel/glitch.hpp

```diff
@@ -0,0 +1,1133 @@
+        RpptROI roi;
+        RpptROIPtr roiPtrInput = &roiTensorPtrSrc[batchCount];
+        compute_roi_validation_host(roiPtrInput, &roi, &roiDefault, roiType);
+
+        Rpp32u xOffsetRchn = xOffsetR[batchCount];
+        Rpp32u yOffsetRchn = yOffsetR[batchCount];
+        Rpp32u xOffsetGchn = xOffsetG[batchCount];
+        Rpp32u yOffsetGchn = yOffsetG[batchCount];
+        Rpp32u xOffsetBchn = xOffsetB[batchCount];
+        Rpp32u yOffsetBchn = yOffsetB[batchCount];
+
+        Rpp32u elementsInRowMax = srcDescPtr->w;
+
+        Rpp32u xOffsets[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsets[3] = {yOffsetRchn, yOffsetGchn, yOffsetBchn};
+        Rpp32u xOffsetsLoc[3] = {xOffsetRchn, xOffsetGchn, xOffsetBchn};
+        Rpp32u yOffsetsLoc[3] = {yOffsetRchn * elementsInRowMax, yOffsetGchn * elementsInRowMax, yOffsetBchn * elementsInRowMax};
+
+        Rpp8u *srcPtrImage, *dstPtrImage;
+        srcPtrImage = srcPtr + batchCount * srcDescPtr->strides.nStride;
+        dstPtrImage = dstPtr + batchCount * dstDescPtr->strides.nStride;
+
+        Rpp32u bufferLength = roi.xywhROI.roiWidth * layoutParams.bufferMultiplier;
+        Rpp8u *srcPtrChannel, *dstPtrChannel;
+        srcPtrChannel = srcPtrImage + (roi.xywhROI.xy.y * srcDescPtr->strides.hStride) + (roi.xywhROI.xy.x * layoutParams.bufferMultiplier);
+        dstPtrChannel = dstPtrImage;
+
+        if ((srcDescPtr->c == 3) && (srcDescPtr->layout == RpptLayout::NHWC) && (dstDescPtr->layout == RpptLayout::NCHW))
+        {
+            convert_pkd3_pln3<Rpp8u>(srcPtrImage, dstPtrImage, srcDescPtr, dstDescPtr, roi.xywhROI.roiWidth * 3);
+            Rpp32u vectorIncrement = 32;
+            for(int c = 0; c < 3; c++)
+            {
+                Rpp32u offset = yOffsetsLoc[c] + xOffsetsLoc[c];
+                Rpp8u *srcPtrImageTemp, *dstPtrImageTemp;
+                srcPtrImageTemp = dstPtrChannel + (c * dstDescPtr->strides.cStride) + offset;
+                dstPtrImageTemp = dstPtrImage + (c * dstDescPtr->strides.cStride);
+                if(offset > 0)
+                {
+                    for(int j = yOffsets[c]; j < roi.xywhROI.roiHeight; j++)
+                    {
+                        Rpp8u *srcPtrTemp = srcPtrImageTemp;
+                        Rpp8u *dstPtrTemp = dstPtrImageTemp;
+                        Rpp32u aligLen = ((roi.xywhROI.roiWidth - xOffsetsLoc[c]) / vectorIncrement) * vectorIncrement;
+                        Rpp32u remElemLen = roi.xywhROI.roiWidth - ( aligLen + xOffsetsLoc[c]);
+                        if(offset < dstDescPtr->strides.cStride)
+                        {
+                            for(int i = 0; i < aligLen; i += vectorIncrement)
+                            {
+                                __m256i p;
+                                p = _mm256_loadu_si256((__m256i *)srcPtrTemp);
+                                _mm256_storeu_si256((__m256i *)dstPtrTemp, p);
```

Comment:
Do we need 2 loads and 2 stores per vector? I think we should discuss this approach